packages/react-components/react-teaching-popover-preview/src/components/TeachingPopover/useTeachingPopover.ts

```diff
@@ -0,0 +1,56 @@
+import { usePopover_unstable } from '@fluentui/react-popover';
+import type { TeachingPopoverProps, TeachingPopoverState } from './TeachingPopover.types';
+import * as React from 'react';
+import { TeachingPopoverCarousel } from '../TeachingPopoverCarousel/TeachingPopoverCarousel';
+
+export function countCarouselChildren(children: React.ReactElement | React.ReactElement[]): number {
+  const childArray = Array.isArray(children) ? children : [children];
+  let childCount = 0;
+  childArray.forEach((child: React.ReactElement) => {
+    if (!child || !child.props) {
+      return;
+    }
+    if (child.props && child.type === TeachingPopoverCarousel) {
+      childCount = React.Children.toArray(child.props.children).length;
+    } else if (child.props && child.props.children) {
+      const newChildCount = countCarouselChildren(child.props.children);
+      if (newChildCount > 0) {
+        childCount = newChildCount;
+      }
+    }
+  });
+
+  return childCount;
+}
+
+export const useTeachingPopover_unstable = (props: TeachingPopoverProps): TeachingPopoverState => {
+  const popoverState = usePopover_unstable(props);
+  const { onFinish, onPageChange } = props;
+
+  const carouselPageCount = React.useCallback(() => {
+    /* We intentionally run this once prior to render
+     * Distinguishes between carousel vs non-carousel visual state
+     * Done prior to initial render to prevent UI change post-render
+     */
+    return countCarouselChildren(props.children);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```

Comment:
Please find another approach to count pages as children manipulation (i.e. `React.Children.*`) never works well with React for shared UI components. For example:

```tsx
<TeachingPopover>
  <Pages /> {/* returns 1, can be 5 */}
</TeachingPopover>
<TeachingPopover>
  <> {/* returns 1, actually is 3 */}
    <div />
    <div />
    <div />
  </>
</TeachingPopover>
```

Reply:
Discussed offline:
Going to move this to a separate task as we don't currently have a pattern here, for now, guidance is to ensure that children are provided as an array of pages or components for pagination.

Task filed under: https://github.com/microsoft/fluentui/issues/26639